components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -217,6 +218,9 @@ public ServiceProvider getApplicationExcludingFileBasedSPs(String applicationNam
     public void updateApplication(ServiceProvider serviceProvider, String tenantDomain, String username)
             throws IdentityApplicationManagementException {
 
+        if (applicationMgtValidator.validateSPConfigurations(serviceProvider, tenantDomain, username)) {
+            throw new IdentityApplicationManagementException("There are application management validation errors");
```

Comment:
No need to throw a new IdentityApplicationManagementException here, since you have added the IdentityApplicationManagementException in method signature, it'll throw the exceptions from validateSPConfigurations() to the upper layer.